examples/loader-google-bigquery/src/index.md

```diff
@@ -0,0 +1,168 @@
+
+```js run=false
+import { csvFormat, csvParse } from "d3-dsv";
+import { runQuery } from "./google-bigquery.js";
+import { promises as fs } from "fs";
+import { dirname, join } from "path";
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const query = `
+  SELECT 
+    FORMAT_TIMESTAMP('%Y-%m-%d', date) as formatted_date, 
+    confirmed_cases 
+  FROM 
+    \`bigquery-public-data.covid19_italy.data_by_province\` 
+  WHERE 
+    name = "Lombardia"
+    AND province_name = "Lecco"
+    AND date BETWEEN '2020-05-01 00:00:00 UTC' AND '2020-05-15 00:00:00 UTC'
+  GROUP BY 1,2
+  ORDER BY 1 ASC;
+`;
+
+(async () => {
+  try {
+    const rows = await runQuery(query);
+
+    console.log("Query Results:", rows);
+
+    if (rows.length === 0) {
+      console.log("No data returned from the query.");
+      return;
+    }
+
+    const csvData = csvFormat(
+      rows.map(d => ({
+        formatted_date: d.formatted_date,
+        confirmed_cases: d.confirmed_cases
+      }))
+    );
+
+    const filePath = join(__dirname, "../covidstats_it.csv");
+    await fs.writeFile(filePath, csvData);
+    console.log(`CSV file written successfully to ${filePath}`);
+
+    const fileContent = await fs.readFile(filePath, "utf8");
+    const parsedData = csvParse(fileContent, { typed: true });
+
+    console.log("Parsed Data:", parsedData);
+
+  } catch (error) {
+    console.error('Error running query or processing file:', error);
+  }
+})();
+```
+
+<div class="tip">
+
+The Google BigQuery API returns dates in ISO 8601 `YYYY-MM-DD` format, which ensures a standard representation that will be correctly parsed by `FileAttachment.csv`.
+
+</div>
+
+The data loader uses a helper file, `google-bigquery.js`, which is a thin wrapper on the `@google-cloud/bigquery` package. This reduces the amount of boilerplate you need to run a report.
+
+```js run=false
+import "dotenv/config";
+import {BigQuery} from "@google-cloud/bigquery";
+
+const {BQ_PROJECT_ID, BQ_CLIENT_EMAIL, BQ_PRIVATE_KEY} = process.env;
+
+if (!BQ_PROJECT_ID) throw new Error("missing BQ_PROJECT_ID");
+if (!BQ_CLIENT_EMAIL) throw new Error("missing BQ_CLIENT_EMAIL");
+if (!BQ_PRIVATE_KEY) throw new Error("missing BQ_PRIVATE_KEY");
+
+const bigQueryClient = new BigQuery({
+  projectId: BQ_PROJECT_ID,
+  credentials: {
+    client_email: BQ_CLIENT_EMAIL,
+    private_key: BQ_PRIVATE_KEY.replace(/\\n/g, '\n')
+  }
+});
+
+export async function runQuery(query) {
+  const [rows] = await bigQueryClient.query({ query });
+  return rows;
+}
+```
+
+<div class="note">
+
+To run this data loader, you’ll need to install `@google-cloud/bigquery`, `d3-dsv`, `d3-time-format`, and `dotenv` using your preferred package manager such as npm or Yarn.
```

Comment:
```suggestion
This data loader depends on `@google-cloud/bigquery`, `d3-dsv`, `d3-time-format`, and `dotenv`, which we reference in `package.json`.
```

I wonder if this kind of wording might be better? Since there's nothing special to do for the user besides installing the project.